eventlet/greenio/py3.py

```diff
@@ -98,7 +101,8 @@ def readall(self):
             except OSError as e:
                 if get_errno(e) not in SOCKET_BLOCKING:
                     raise IOError(*e.args)
-            self._trampoline(self, read=True)
+            if not self._is_regular:
```

Comment:
According to `stat()` documentation, file descriptor may be a regular file, directory, symlink, block or char device, socket and fifo. What are supported types? Is it possible to use white list instead? Check that fd is of supported type.